pkg/kubelet/allocation/state/state_mem.go

```diff
@@ -77,17 +90,41 @@ func (s *stateMemory) SetContainerResources(podUID types.UID, containerName stri
 	s.Lock()
 	defer s.Unlock()
 
-	if _, ok := s.podResources[podUID]; !ok {
-		s.podResources[podUID] = PodResourceInfo{
+	podInfo, ok := s.podResources[podUID]
+	if !ok {
+		podInfo = PodResourceInfo{
 			ContainerResources: make(map[string]v1.ResourceRequirements),
```

Comment:
Are these lines necessary? The returned podInfo from `podInfo, ok := s.podResources[podUID]` should be a `PodResourceInfo{}`, so L100-102 should cover the `!ok` case